src/js/game/hud/parts/building_placer.js

```diff
@@ -105,6 +107,49 @@ export class HUDBuildingPlacer extends BaseHUDPart {
         }
     }
 
+    usePippete() {
+        if (this.currentMetaBuilding.get()) {
+            return;
+        }
+        if (this.root.camera.getIsMapOverlayActive()) {
+            return;
+        }
+
+        const mousePos = this.root.app.mousePosition;
+        if (!mousePos) {
+            return;
+        }
+        const worldPos = this.root.camera.screenToWorld(mousePos);
+        const worldTile = worldPos.toTileSpace();
+
+        const entity = this.root.map.getTileContent(worldTile);
+        if (!entity) {
+            return;
+        }
+        const staticEntity = entity.components.StaticMapEntity;
+
+        this.currentBaseRotation = (Math.round(staticEntity.originalRotation / 90) * 90 + 360) % 360;
+        // maybe not originalRotation, but rotation
+
+        if (!staticEntity.spriteKey) {
+            if (entity.components.Belt) {
+                this.currentMetaBuilding.set(gMetaBuildingRegistry.findById("belt"));
+            }
+            return;
+        }
+
+        let match = staticEntity.spriteKey.match(
+            /sprites\/buildings\/(.*?)(_entry|_exit|_[a-z]+|)(|-(.*))\.png/
+        );
+        const id = match[1];
+        const variant = match[4] || defaultBuildingVariant;
```

Comment:
If I understood correctly, buildings do not store their ids and variants

Reply:
Oof, this is a pretty hacky solution. I'd prefer to find a better solution before mergin this